nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,867 @@
+
+        if is_classif:
+            if hasattr(estimator, 'predict_proba'):
+                y_prob = estimator.predict_proba(X_test)
+                y_prob = y_prob[:, 1]
+                neg_prob = 1 - y_prob  # the complement of the probability
+            else:
+                decision = estimator.decision_function(X_test)
+                if decision.ndim == 2:
+                    y_prob = decision[:, 1]
+                    neg_prob = np.abs(decision[:, 0])
+                else:
+                    y_prob = decision
+                    neg_prob = -decision  # negative decision function
+            score = scorer(estimator, X_test, y_test)
+            if np.all(estimator.coef_ == 0):
+                score = 0
+        else:  # regression
+            y_prob = estimator.predict(X_test)
+            score = scorer(estimator, X_test, y_test)
+
+        if (best_score is None) or (score >= best_score):
+            best_score = score
+            best_coef = estimator.coef_
+            best_intercept = estimator.intercept_
+            best_y = {}
+            best_y['y_prob'] = y_prob
+            best_y['y_true_indices'] = test
+            best_param = param
+            if is_classif:
+                best_y['inverse'] = neg_prob
+
+    if (selector is not None) and do_screening:
+        best_coef = selector.inverse_transform(best_coef)
+
+    return (class_index, best_coef, best_intercept, best_y, best_param,
+            best_score)
+
+
+class BaseDecoder(LinearModel, RegressorMixin, CacheMixin):
+    """A wrapper for popular classification/regression strategies in
+    neuroimaging.
+
+    The `BaseDecoder` object supports classification and regression methods.
+    It implements a model selection scheme that averages the best models
+    within a cross validation loop. The resulting average model is the
+    one used as a classifier or a regressor. This object also leverages the
+    `NiftiMaskers` to provide a direct interface with the Nifti files on disk.
+
+    Parameters
+    -----------
+    estimator : str, optional
+        The estimator to choose among: 'svc', 'svc_l2', 'svc_l1', 'logistic',
+        'logistic_l1', 'logistic_l2', 'ridge', 'ridge_classifier',
+        'ridge_regressor', and 'svr'. Note that the 'svc' and 'svc_l2';
+        'logistic' and 'logistic_l2'; 'ridge' and 'ridge_classifier' correspond
+        to the same estimator. Default 'svc'.
+
+    mask : filename, Nifti1Image, NiftiMasker, or MultiNiftiMasker, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask and parameters will be used. If no mask is given, mask
+        will be computed automatically from provided images by an inbuilt
+        masker with default parameters. Refer to NiftiMasker or
+        MultiNiftiMasker to check for default parameters. Default None
+
+    cv : cross-validation generator or int, optional. Default 10
+        A cross-validation generator. If None, a 3-fold cross
```

Comment:
The "None" behavior is changing in scikit-learn. I would just avoid mentioning it.
```suggestion
        A cross-validation generator.
```